# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.h
# Commit: dbdca4661a35
# Full Hash: dbdca4661a35c55d343cb76fd10366cdbd1e4c09
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r ea27cd66fefd -r dbdca4661a35 dom/fetch/InternalResponse.h
--- a/dom/fetch/InternalResponse.h	Tue Jan 17 15:13:44 2023 +0000
+++ b/dom/fetch/InternalResponse.h	Tue Jan 17 15:13:45 2023 +0000
@@ -57,6 +57,9 @@
 
   ParentToParentInternalResponse ToParentToParentInternalResponse();
 
+  ParentToChildInternalResponse ToParentToChildInternalResponse(
+      NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent);
+
   enum CloneType {
     eCloneInputStream,
     eDontCloneInputStream,
@@ -337,6 +340,8 @@
 
   SafeRefPtr<InternalResponse> Unfiltered();
 
+  InternalResponseMetadata GetMetadata();
+
   ~InternalResponse();
 
  private:
@@ -351,8 +356,6 @@
   template <typename T>
   static SafeRefPtr<InternalResponse> FromIPCTemplate(const T& aIPCResponse);
 
-  InternalResponseMetadata GetMetadata();
-
   ResponseType mType;
   // A response has an associated url list (a list of zero or more fetch URLs).
   // Unless stated otherwise, it is the empty list. The current url is the last